Backend/src/KCTest.Application/Services/PermissionService.cs

```diff
@@ -74,12 +75,20 @@ public async Task<PermissionDto> GetPermission(int permissionId)
             return permissionDto;
         }
 
-        public async Task<IEnumerable<PermissionDto>> GetPermissions(Pagination pagination = null)
+        public async Task<IEnumerable<PermissionDto>> GetPermissions()
         {
             var permissions = await _unitOfWork.PermissionRepository.GetAllAsync(entitiesToInclude: new List<string> { "PermissionType" });
             var permissionsDto = _mapper.Map<IEnumerable<PermissionDto>>(permissions);
 
             return permissionsDto;
         }
+
+        public async Task<IEnumerable<PermissionDto>> GetPermissions(Pagination pagination)
+        {
+            var permissions = await _unitOfWork.PermissionRepository.GetAllAsync(entitiesToInclude: new List<string> { "PermissionType" });
+            var permissionsDto = _mapper.Map<IEnumerable<PermissionDto>>(permissions);
+
+            return permissionsDto.Skip(pagination.Skip).Take(pagination.Limit);
```

Comment:
Have in mind that here you're paginating after getting the results from the repository. Most efficient way is that the repository paginates the results. So that if you're querying a 1 million rows table, you'll not be getting 1 million rows and then paginate